### PR TITLE
Add option to open command results in a new tab

### PR DESCRIPTION
**Pull request by mburleigh**
The Rest Client extension (https://marketplace.visualstudio.com/items?itemName=humao.rest-client) has functionality like this and after using the Azure CLI Tools extension for a while I missed being able to have the results of multiple commands available. This will open a new results tab every time "Run Line in Editor" is executed.

### CODE REVIEW

**Comment by mburleigh** [comment]
@chrisdias @chrmarti is there anything I can do to move this PR along?

**Comment by chrmarti** [comment]
@mburleigh Sorry, we were heads-down into another part of the VS Code project. I'll take a look.

File: src/extension.ts [hunk 1]
```diff
@@ -188,6 +205,10 @@ class RunLineInEditor {
     }
 
     private findResultDocument() {
+        if (this.settings.showResponseInDifferentTab) {
```
Comment (chrmarti) [inline src/extension.ts]: To keep the change simple, just read the setting with workspace.getConfiguration() on line 212 below: `if (this.resultDocument && !workspace.getConfiguration(...)) {`

File: src/extension.ts [hunk 2]
```diff
@@ -147,35 +146,53 @@ class RunLineInTerminal {
...
+        this.interval = setInterval(() => {
+            this.runStatusBarItem.text = `Waiting for response ${this.spinner()}`;
+        }, 50);
+        this.runStatusBarItem.show();
+
         this.parsedResult = undefined;
         this.query = undefined; // TODO
         const cursor = source.selection.active;
         const line = source.document.lineAt(cursor).text;
+        const isPlainText = (line.indexOf('--query') !== -1) || (line.indexOf('-h') !== -1) || (line.indexOf('--help') !== -1);
```
Comment (chrmarti) [inline src/extension.ts]: Could you do this in a separate PR? And maybe check the output if it starts and ends with `{}` or `[]` to assume it is JSON. That would handle more cases.

File: src/extension.ts [hunk 3]
```diff
@@ -147,35 +146,53 @@ class RunLineInTerminal {
...
+    private interval!: NodeJS.Timer;
+    private spinner = elegantSpinner();
 
     constructor(private status: StatusBarInfo) {
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.toggleLiveQuery', editor => this.toggleQuery(editor)));
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.runLineInEditor', editor => this.run(editor)));
         this.disposables.push(workspace.onDidCloseTextDocument(document => this.close(document)));
         this.disposables.push(workspace.onDidChangeTextDocument(event => this.change(event)));
+
+        this.runStatusBarItem = window.createStatusBarItem(StatusBarAlignment.Left);
```
Comment (chrmarti) [inline src/extension.ts]: Could you do this in a separate PR? And maybe hide the status bar item 10 seconds after the command finished, so it doesn't use up space in the status bar when it is no longer needed.

File: src/extension.ts [hunk 4]
```diff
@@ -304,8 +326,11 @@ function allMatches(regex: RegExp, string: string, group: number) {
     }
 }
 
-function replaceContent(editor: TextEditor, content: string) {
+function replaceContent(editor: TextEditor, content: string, documentLanguage: string = '') {
     const document = editor.document;
+    if (documentLanguage) {
```
Comment (chrmarti) [inline src/extension.ts]: It would be better to always update because in the case where the document is reused, it might receive JSON again after plain text.

File: src/extension.ts (hunk 1 above)
Comment (mburleigh) [reply src/extension.ts]: @chrmarti - I'm going to abandon this PR and split the code 3 smaller ones (tabs, doc language, status bar item) based on your comments.

Comment (mburleigh) [reply src/extension.ts]: Functionality moved to https://github.com/microsoft/vscode-azurecli/pull/55

File: src/extension.ts (hunk 3 above)
Comment (mburleigh) [reply src/extension.ts]: This has been separated into https://github.com/microsoft/vscode-azurecli/pull/56

**Comment by chrmarti** [comment]
Closing this PR as you split out separate ones.
